### PR TITLE
[MB-15297] Upgrade to go 1.20.1

### PR DESCRIPTION
**Pull request by reggieriser**
## Summary

This PR updates the project from Go 1.19.3 to Go 1.20.1 (which includes security fixes and some possible options related to FIPS 140-2 compliance).

Go 1.20.x version history:
https://golang.org/doc/devel/release.html#go1.20

Go 1.19.x version history:
https://golang.org/doc/devel/release.html#go1.19

Here's the [corresponding PR from circleci-docker](https://github.com/transcom/circleci-docker/pull/381) for the docker image hash referenced in this PR.  Please verify that it seems correct and that all relevant references have been updated.

~~**Note:** After this PR has been initially vetted, I will merge in the circleci-docker PR and update this PR with the new hash.~~ DONE!

I generally followed [these upgrade instructions](https://transcom.github.io/mymove-docs/docs/backend/guides/how-to/upgrade-go-version).

Some issues that were discovered by the new golangci-lint were already fixed in [this earlier merged PR](https://github.com/transcom/mymove/pull/10034).

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

If you're on `asdf`:
- `asdf install`
- `asdf global golang 1.20.1`

If you're on `nix`:
- run `./nix/update.sh`

Just verify that all tests still pass and that your normal workflow seems to be working fine.  Let me know if you're using `asdf` or `nix` -- I'd like to have at least one reviewer from each.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: .github/workflows/go-auto-approve.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
       - name: setup go
         uses: actions/setup-go@v3
         with:
-          go-version: '1.19'
+          go-version: '1.20.1'
```
Comment (reggieriser) [inline .github/workflows/go-auto-approve.yml:30]: Historically, we've only put the major version here, but based on what I see [in the docs for `setup-go`](https://github.com/actions/setup-go), I think that imples we are only using the initial release and not any patches.  So, I'm changing this to match our specific go version.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@ repos:
         exclude: ^public/swagger-ui/
 
   - repo: https://github.com/golangci/golangci-lint
-    rev: v1.50.1
+    rev: v1.51.2
```
Comment (reggieriser) [inline .pre-commit-config.yaml:53]: We have to upgrade `golangci-lint` to get compatibility with go 1.20.

**Review by ahobson — APPROVED** [review]
Using nix, I ran `make server_test` and `precommit run -a` and both were fine after installing go via nix.

**Comment by duncan-truss** [comment]
I'm seeing that the `-i` command being passed to `go build` has been removed after being deprecated for a while.

Screenshot 2023-02-28 at 3 47 07 PM

This is what I get when running `make server_generate && make server_run`:

```sh
flag provided but not defined: -i
usage: go build [-o output] [build flags] [packages]
Run 'go help build' for details.

make[1]: *** [server_run_default] Terminated: 15
```

**Comment by reggieriser** [comment]
> I'm seeing that the `-i` command being passed to `go build` has been removed after being deprecated for a while.

Good catch, @duncan-truss !  I've removed that deprecated/removed flag and `server_run` seems to start up without errors now.

File: .github/workflows/go-auto-approve.yml (hunk 1 above)
Comment (felipe-lee) [reply .github/workflows/go-auto-approve.yml:30]: **praise**: nice catch

**Review by felipe-lee — APPROVED** [review]
Lgtm (as a `nix` user)! I ran `make server_test` and `make server_run` (with `make client_run`) and things seem to work without issues.

**Review by Ryan-Koch — APPROVED** [review]

**Review by duncan-truss — APPROVED** [review]

**Pull request merged**